### ISSUE TITLE
Example of best practice (Async) Routes in a separate file

### ISSUE THREAD

**Issue by calumk** [talentlessguy/tinyhttp#200]
**Is your feature request related to a problem? Please describe.**
Could you please provide an Example of placing (Async) routes in a separate file

**Describe the solution you'd like**
I am unable to successfully include a route with async functions in a separate file.

The following - Works perfectly in the main file.

```
app.get('/asyncTest',  (req, res) => {
   res.render('index.eta', { 
   name: 'Calum',
   af : asyncFunctions
 });
});
```

The following - fails if placed in a separate routes.js file. (route containing non-async works)

```
router.get('/asyncTest',  (req, res) => {
   res.render('index.eta', { 
   name: 'Calum',
   af : asyncFunctions
 });
});
```
attached with `app.use(router)`,

Warning is `Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`

**Additional context**
If there is any chance of adding an example showing how to do this, that would be appreciated 👍

**Comment by talentlessguy** [comment]
how did you create the router? did you do `const router = new App()`?

**Comment by calumk** [comment]
Code is CommonJS not EMS, (although Originally i had it written in EMS and experienced identical behaviour)

asynchronous functions are declared separately in the asyncFunctions object,  and passed to eta.
https://eta.js.org/docs/syntax/async

This works in main.js, but doesn't work if i specify it in a separate file.

I have simplified my code but below should explain


**asyncFunctions.js**
```
let asyncFunctions = {
    asyncFunc : () => {
        return new Promise(resolve => {
            setTimeout(() => {
                resolve('HI FROM ASYNC')
            }, 10)
        })
    }
}

module.exports = asyncFunctions;
```


### Working
**main.js**
```
let tinyHttp = require('@tinyhttp/app');
let app = new tinyHttp.App()
let asyncFunctions = require('asyncFunctions.js');

app.engine('eta', Eta.renderFileAsync);

app.get('/', (req, res) => {
    res.render('index.eta', { 
     name: 'Calum',
     af : asyncFunctions
  });
 });

 app.listen(1337, () => console.log(`Listening on http://localhost:1337`))
```


---
### Not Working
_Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client_

**Main.js**
```
let tinyHttp = require('@tinyhttp/app');
let app = new tinyHttp.App()
let custom_routes = require('routes.js');

app.engine('eta', Eta.renderFileAsync);

app.use(custom_routes)

app.listen(1337, () => console.log(`Listening on http://localhost:1337`))
```

**routes.js**
```
let tinyHttp = require('@tinyhttp/app');
let Eta = require("eta");
let asyncFunctions = require('asyncFunctions.js');

let routes = new tinyHttp.App()
routes.engine('eta', Eta.renderFileAsync);

routes.get('/',  (req, res) => {
   res.render('index.eta', { 
   name: 'Calum',
   af : asyncFunctions
 });
});

module.exports = routes
```

**Comment by talentlessguy** [comment]
what is the contents of `routes.js` file?

**Comment by talentlessguy** [comment]
@calumk from the docs it says that you must enable `async: true` as a template option.

To do that, you need to pass settings to `res.render`:

**typescript version**

```ts
import { App } from '@tinyhttp/app'
import { renderFile as eta } from 'eta'
import { EtaConfig } from 'eta/dist/types/config'

const app = new App<EtaConfig>()

app.engine('eta', eta)

function func() {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve('HI FROM ASYNC')
    }, 20)
  })
}

app.use(
  (_, res) =>
    void res.render(
      'index.eta',
      { name: 'Eta', func },
      {
        renderOptions: {
          async: true
        }
      }
    )
)

app.listen(3000, () => console.log(`Listening on http://localhost:3000`))
```

**javascript version**:

```js
import { App } from '@tinyhttp/app'
import { renderFile as eta } from 'eta'

const app = new App()

app.engine('eta', eta)

function func() {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve('HI FROM ASYNC')
    }, 20)
  })
}

app.use(
  (_, res) =>
    void res.render(
      'index.eta',
      { name: 'Eta', func },
      {
        renderOptions: {
          async: true
        }
      }
    )
)

app.listen(3000, () => console.log(`Listening on http://localhost:3000`))
```

**Comment by talentlessguy** [comment]
ok looks like there is a router bug, it doesn't seem to queue wares properly 

will look into that

**Comment by talentlessguy** [comment]
Fixed in v1.0.13:

```sh
pnpm i @tinyhttp/app@1.0.13
```

**Comment by calumk** [comment]
@talentlessguy , Many thanks for your work on this, I really appreciate it, and how quick you were.

I can confirm - 1.0.13 fixed the issue without the need for me to re-write anything.

{async: true} is not required, due to new sugar in Eta -> Eta.renderFileAsync 
https://github.com/eta-dev/eta/issues/49
